contracts/ProposalsContract.sol

```diff
@@ -0,0 +1,154 @@
+	 * @param _multisigAddress – address of the mulitisig contract (that controls us)
+	 */
+	constructor(PreserveBalancesOnTransferToken _token, address _multisigAddress) public {
+		multisigAddress = _multisigAddress;
+		token = _token;
+	}
+
+	/**
+	 * @dev Propose to change _paramType value to _param
+	 * @notice This function creates proposal
+	 * @param _target – target of call
+	 * @param _data – call data
+	 */
+	function propose(address _target, bytes _data) onlyMultisigAddress public {
+		Proposal memory p;
+		p.startedAt = now;
+		p.target = _target;
+		p.data = _data;
+		p.eventId = token.startNewEvent();
+		p.vetoScore = 0;
+		p.totalSupplyAtEvent = token.totalSupply();
+		proposals.push(p);	
+		emit ProposalStarted(_target, _data, p.totalSupplyAtEvent, p.eventId, msg.sender);	
+	}
+
+	/**
+	 * @dev Get ALL proposal count 
+	 * @notice This function can be called by anyone
+	 * @return Proposal count
+	 */
+	function getProposalsCount() public view returns(uint) {
+		return proposals.length;
+	}
+
+	/**
+	* @dev Get proposal data
+	 * @notice This function can be called by anyone
+	 * @param _proposalIndex – proposal number
+	 * @return target – address of a contract with the function, that will be called if proposal will be accepted
+	 * @return data – data of the function, that will be called if proposal will be accepted
+	 * @return paramValue – what is param amount
+	 * @return pro – sum of voters token amount, that voted yes
+	 * @return vetoScore – sum of voters token amount, that voted no
+	 * @return isFinished – is Quorum reached
+	 * @return isVetoed – is veto percent > 30
+	 */
+	function getProposalStats(uint _proposalIndex) public view returns(address target, bytes data, bytes32 paramValue, uint pro, uint vetoScore, bool isFinished, bool isVetoed) {
+		require(_proposalIndex<proposals.length);
+		
+		target = proposals[_proposalIndex].target;
+		data = proposals[_proposalIndex].data;		
+		vetoScore = proposals[_proposalIndex].vetoScore;
+		isFinished = _isProposalFinished(_proposalIndex);
+		isVetoed = _isVetoed(_proposalIndex);
+	}
+
+	/**
+	 * @dev Is selected proposal finished?
+	 * @param _proposalIndex – proposal number
+	 * @return is proposal finished or not
+	 */
+	function _isProposalFinished(uint _proposalIndex) internal view returns(bool isIt) {
+		require(_proposalIndex<proposals.length);
+		
+		if (((now) >= (proposals[_proposalIndex].startedAt + daysToVote * 1 days)) || 
+			(_isVetoed(_proposalIndex))) {
+			isIt = true;
+		}
+	}
+
+	/**
+	 * @dev Is selected proposal vetoed?
+	 * @param _proposalIndex – proposal number
+	 * @return is proposal vetoed or not
+	 */	
+	function _isVetoed(uint _proposalIndex) internal view returns(bool isIt) {
+		if((proposals[_proposalIndex].vetoScore*100) >= (proposals[_proposalIndex].totalSupplyAtEvent*VETO_PERCENT)) {
+			isIt = true;
+		}
+	}
+
+	function finalize(uint _proposalIndex) public {
+		require(!proposals[_proposalIndex].isFinalized);
```

Comment:
please add messages for all the `require`s, so we now the reason for reverts if any